app/scripts/controllers/selfservice/CreateSelfServiceUserController.js

```diff
@@ -0,0 +1,98 @@
+(function (module) {
+    mifosX.controllers = _.extend(module, {
+        CreateSelfServiceUserController: function (scope, resourceFactory, routeParams,$uibModal) {
+            // TODO: Add Account Is Active Logic through email validation
+            scope.accountActive = false;    
+            scope.available = [];
+            scope.selected = [];
+            scope.selectedRoles = [] ;
+            scope.availableRoles = [];
+            scope.formData = {
+                isSelfServiceUser: true,
+                sendPasswordToEmail: true,
+                roles: []
+            };
+            scope.clientId = routeParams.clientId;
+            resourceFactory.userTemplateResource.get(function (data) {
+                scope.availableRoles = data.availableRoles;
```

Comment:
@nazeer1100126  Yes I did have doubts on this, What roles are the valid role for creation of the self-service user? 
How do I fetch roles specific for creating the self-service user? 
Can we update the template on the backend for normal user creation and a self-service user creation with appropriate roles?